symplyphysics/laws/thermodynamics/zero_heat_transfer.py

```diff
@@ -25,26 +41,119 @@
 adiabatic_condition = Eq(pressure_start * (volume_start**specific_heats_ratio),
     pressure_end * (volume_end**specific_heats_ratio))
 
-eq_start = thermodynamics_law.law.subs({
-    thermodynamics_law.temperature: temperature_start,
-    thermodynamics_law.volume: volume_start,
-    thermodynamics_law.pressure: pressure_start
+eq_start = ideal_gas_equation.law.subs({
+    ideal_gas_equation.temperature: temperature_start,
+    ideal_gas_equation.volume: volume_start,
+    ideal_gas_equation.pressure: pressure_start
 })
 
-eq_end = thermodynamics_law.law.subs({
-    thermodynamics_law.temperature: temperature_end,
-    thermodynamics_law.volume: volume_end,
-    thermodynamics_law.pressure: pressure_end
+eq_end = ideal_gas_equation.law.subs({
+    ideal_gas_equation.temperature: temperature_end,
+    ideal_gas_equation.volume: volume_end,
+    ideal_gas_equation.pressure: pressure_end
 })
 
 law = [eq_start, eq_end, adiabatic_condition]
 
+# Derive from first law of thermodynamics and other relations
+
+_temperature = ideal_gas_equation.temperature
+_pressure = ideal_gas_equation.pressure
+_volume = ideal_gas_equation.volume
+
+_adiabatic_index = Symbol("adiabatic_index", positive=True)
+_isobaric_heat_capacity = mayers_relation.isobaric_heat_capacity
+_isochoric_heat_capacity = mayers_relation.isochoric_heat_capacity
+
+_temperature_change = SymSymbol("temperature_change")
+_pressure_change = SymSymbol("pressure_change")
+_volume_change = SymSymbol("volume_change")
+
+_internal_energy_expr = internal_energy_law.law.rhs.subs({
+    internal_energy_law.temperature: _temperature,
+    internal_energy_law.isochoric_heat_capacity: _isochoric_heat_capacity,
+})
+
+_internal_energy_change_expr = _internal_energy_expr.diff(_temperature) * _temperature_change
```

Comment:
Here it assumes that heat capacity is constant. But generally it is a function of temperature.

Reply:
True. I should refactor that law so that it has differentials instead of integrated quantities and this proof wouldn't need an unnecessary conditon. Originally it is $`\mathrm{d}U = C_V \mathrm{d}T`$.

Reply:
Opened #679

Reply:
Meged